pkg/cmd/server/origin/run_components.go

```diff
@@ -163,7 +165,19 @@ func (c *MasterConfig) RunServiceAccountTokensController(cm *cmapp.CMServer) {
 // RunServiceAccountPullSecretsControllers starts the service account pull secret controllers
 func (c *MasterConfig) RunServiceAccountPullSecretsControllers() {
 	serviceaccountcontrollers.NewDockercfgDeletedController(c.KubeClientset(), serviceaccountcontrollers.DockercfgDeletedControllerOptions{}).Run()
-	serviceaccountcontrollers.NewDockercfgTokenDeletedController(c.KubeClientset(), serviceaccountcontrollers.DockercfgTokenDeletedControllerOptions{}).Run()
+
+	ctx := origincontroller.ControllerContext{
```

Comment:
This block has the right idea.  Please pull it out into a method which constructs the context and initializers and starts them all.  Having only one in the map (as you have now) is perfect.